provisioner/ibm-s3fs-provisioner.go

```diff
@@ -399,7 +441,65 @@ func (p *IBMS3fsProvisioner) Provision(options controller.VolumeOptions) (*v1.Pe
 		}
 	}
 
+	contextLogger.Info(pvcName + ":" + clusterID + " ConfigBucketAccessPolicy: " + strconv.FormatBool(*ConfigBucketAccessPolicy) + ", SetAccessPolicy: " + pvc.SetAccessPolicy)
+
+	//add check for region = BNNP
+	if ConfigBucketAccessPolicy != nil && *ConfigBucketAccessPolicy && pvc.SetAccessPolicy != "false" {
+		grpcSess = p.GRPCBackend.NewGrpcSession()
+		cc := &grpcClient.GrpcSes{}
+		conn, err := grpcSess.GrpcDial(cc, *SockEndpoint, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithDialer(UnixConnect))
+		if err != nil {
+			return nil, fmt.Errorf(pvcName+":"+clusterID+":failed to establish grpc-client connection: %v", err)
+		}
+
+		providerClient = p.IBMProvider.NewIBMProviderClient(conn)
+		if conn != nil {
```

Comment:
Also why we are not calling this section of code for closing connection at line  454?

Reply:
So this call `p.IBMProvider.NewIBMProviderClient` wont fail in any case? if thats the case, then its fine.

Reply:
since `conn` is getting passed while creating IBMProviderClient instance, it would be good practise to call defer after its usage.